source/MaterialXFormat/XmlIo.cpp

```diff
@@ -220,10 +220,13 @@ void documentFromXml(DocumentPtr doc,
                      const FileSearchPath& searchPath = FileSearchPath(),
                      const XmlReadOptions* readOptions = nullptr)
 {
+    FileSearchPath xmlSearchPath(searchPath);
```

Comment:
Going with most common location for env path addition.

Reply:
This doesn't prevent adding the environment path multiple times, since the modified path is still recursively passed through `documentFromXml -> processXIncludes -> readFromXmlFile -> documentFromXml -> ...`. So we don't need to create a copy here (the one in `readFromXmlFile` makes sense).